kore/src/Kore/Step/Simplification/OverloadSimplifier.hs

```diff
@@ -0,0 +1,140 @@
+{- |
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+
+-}
+module Kore.Step.Simplification.OverloadSimplifier
+    ( OverloadSimplifier (..)
+    , mkOverloadSimplifier
+    ) where
+
+import Control.Exception
+    ( assert
+    )
+import Data.Function
+    ( on
+    )
+import qualified Data.Map.Strict as Map
+import Data.Maybe
+    ( listToMaybe
+    , mapMaybe
+    )
+import qualified Data.Set as Set
+import qualified GHC.Stack as GHC
+
+import qualified Kore.Attribute.Axiom as Attribute
+import qualified Kore.Attribute.Symbol as Attribute
+import Kore.Attribute.Synthetic
+    ( synthesize
+    )
+import Kore.IndexedModule.IndexedModule
+    ( VerifiedModule
+    , recursiveIndexedModuleAxioms
+    )
+import Kore.IndexedModule.MetadataTools as MetadataTools
+import Kore.IndexedModule.SortGraph
+    ( SortGraph
+    , isSubsortOf
+    )
+import Kore.Internal.ApplicationSorts
+    ( applicationSortsResult
+    )
+import qualified Kore.Internal.Symbol as Symbol
+import Kore.Internal.TermLike
+
+data OverloadSimplifier =
+    OverloadSimplifier
+        { isOverloading :: Symbol -> Symbol -> Bool
```

Comment:
`isOverloading` and `isOverloaded` are good candidates for extraction into an `OverloadGraph` similar to `SortGraph`, but I don't know that it's essential to do that.